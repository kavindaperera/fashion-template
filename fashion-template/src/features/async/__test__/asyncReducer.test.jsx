import * as types from '../asyncConstants';
import reducer from '../asyncReducer';
import { initialState } from '../asyncReducer';

const trueState = {
  loading: true
}

describe('async reducer', () => {
    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toEqual(
        {
            loading: false
        }
      )
    })
    //initial state to start state
    it('should handle ASYNC_ACTION_START', () => {
        expect(
          reducer(initialState , {
            type: types.ASYNC_ACTION_START
          })
        ).toEqual({
            loading: true
          })

    })
    //initial state to finish state
    it('should handle ASYNC_ACTION_FINISH', () => {
        expect(
          reducer(initialState , {
            type: types.ASYNC_ACTION_FINISH
          })
        ).toEqual({
            loading: false
          })

    })
    //initial state to error state
    it('should handle ASYNC_ACTION_ERROR', () => {
      expect(
        reducer(initialState , {
          type: types.ASYNC_ACTION_ERROR
        })
      ).toEqual({
          loading: false
        })

    })
    //true state to start state
    it('should handle ASYNC_ACTION_START', () => {
      expect(
        reducer(trueState , {
          type: types.ASYNC_ACTION_START
        })
      ).toEqual({
          loading: true
        })

    })
    //true state to finish state
    it('should handle ASYNC_ACTION_FINISH', () => {
      expect(
        reducer(trueState , {
          type: types.ASYNC_ACTION_FINISH
        })
      ).toEqual({
          loading: false
        })

    })
    //true state to error state
    it('should handle ASYNC_ACTION_ERROR', () => {
      expect(
        reducer(trueState , {
          type: types.ASYNC_ACTION_ERROR
        })
      ).toEqual({
          loading: false
        })

    })

    //Snapshot testing

    it('should handle ASYNC_ACTION_START', () => {
      expect(
        reducer(initialState , {
          type: types.ASYNC_ACTION_START
        })
      ).toMatchSnapshot()

    })
    it('should handle ASYNC_ACTION_FINISH', () => {
      expect(
        reducer(initialState , {
          type: types.ASYNC_ACTION_FINISH
        })
      ).toMatchSnapshot()

    })

    it('should handle ASYNC_ACTION_ERROR', () => {
      expect(
        reducer(initialState , {
          type: types.ASYNC_ACTION_ERROR
        })
      ).toMatchSnapshot()

    })

})