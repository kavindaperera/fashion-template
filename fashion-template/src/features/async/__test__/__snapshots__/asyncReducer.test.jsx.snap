// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async reducer should handle ASYNC_ACTION_ERROR 1`] = `
Object {
  "loading": false,
}
`;

exports[`async reducer should handle ASYNC_ACTION_FINISH 1`] = `
Object {
  "loading": false,
}
`;

exports[`async reducer should handle ASYNC_ACTION_START 1`] = `
Object {
  "loading": true,
}
`;
